### ISSUE TITLE
column header and content not aligned

### ISSUE THREAD

**Issue by yaxino** [olifolkerd/tabulator#215]
it is a bug ! Hope to get your answer
when i use setData to draw table. 
then Drag the column width , and  Change column order
then i click search button, call the setData function.
Bug appears!
 
![1](https://cloud.githubusercontent.com/assets/26379258/23850693/79789094-081b-11e7-8b9f-d674d126e9be.png)
![2](https://cloud.githubusercontent.com/assets/26379258/23850700/826c28e6-081b-11e7-8704-d9d12c1f0495.png)
![3](https://cloud.githubusercontent.com/assets/26379258/23850702/86540cf8-081b-11e7-9971-ba743013a3e1.png)

**Comment by olifolkerd** [comment]
Hey,

Thanks for getting in touch,

Sorry to hear you are having issues.

Could you post a copy of your Tabulator constructor object so i can run some tests on a matching table for you.

Cheers

Oli

**Comment by yaxino** [comment]
$(obj.selector + ' .tabulator').tabulator({
                fitColumns: true,
                //pagination: true,
                movableCols: true,
                persistentLayout: true,
                movableRows: true,
                columns: obj.columns,
                renderComplete: function (sortList, dir) {
                    
                }
            });


 Utility.ajax.post(obj.url,obj.postData, function (data) {
                if (!!data) {
                    
                    $(obj.selector + ' .tabulator').tabulator("setData", data);
                    $(obj.selector + ' .tabulator').tabulator("redraw", true);
                }
            }, obj.selector);

**Comment by olifolkerd** [comment]
Hey,

could you include your column definition object as that will play a key role in how the system handles column resizing,

Cheers

Oli

**Comment by yaxino** [comment]
my columns ( By the way, any simple column definition can reproduce the bug. 
1st:Change the width of a column , 
2th:Change column order,
3rd:click search button.
then you can find header and content is not aligned !): 
[
            {
                title: "Cost Center", width: 100, tooltip:function(field, value, data){
                    data = JSON.parse(data.WFJson);
                    var tip = '';
                    if (data.CC) {
                        tip += ' Cost Center:' + data.CC ;
                    }
                    if (data.PC) {
                        tip += ' Project Code:' + data.PC;
                    }
                    return tip; //return cells "field - value";
                }, formatter: function (value, data) {
                    
                    var speedData = JSON.parse( data.WFJson);
                    return "<div>" + '<a href="editcc.html?id=' + data.RequestSN + '&cc=' + (speedData.CC ? speedData.CC : '') + '&pc=' + (speedData.PC ? speedData.PC : '') + '" class="btn btn-xs btn-warning btn-radius"><i class="ace-icon fa fa-pencil-square-o"></i>Edit</a>' + "</div>";
                }
            },
            { title: "ID", field: "RequestSN", width: 120 },
			{ title: "Title", field: "Title", width: 200 },
			{
			    title: "Submiter", field: "ApplyUser", formatter: function (value, data) {

			        return "<div>" + value.FirstName + ' ' + value.LastName + "</div>";
			    }
			},
			{
			    title: "Submit Date", field: "ApplyDate", formatter: function (value, data) {
			        return "<div>" + Utility.convert.toDate(value) + "</div>";
			    }
			}, 
			{
			    title: "Next Reviewer", field: "Checkers", formatter: function (value, data) {
			        var ck = '';
			        if (value) {
			            for (var i in value) {
			                ck += '|<a href="mailto:' + value[i].Email
                                + '?subject=Process Approve: &body=From GPMP System Url:' + window.location.href
                                + '"><b class="blue">' + value[i].FirstName + ' ' + value[i].LastName + '</b></a>'
			            }
			            if (ck.length > 0) {
			                ck += '|';
			            }
			        }
			        return "<div>" + ck + "</div>";
			    }
			},
			{
			    title: "Status", field: "FormState", formatter: function (value, data) {
			        var status = '';
			        switch (value) {
			            case 'Submit':
			                status = '<span class="label label-success arrowed-in arrowed-in-right">Submit</span>'
			                break;
			            case 'Reject':
			                status = '<span class="label label-danger arrowed-left">Reject</span>'
			                break;
			            case 'Complete':
			                status = '<span class="label label-warning arrowed arrowed-right">Complete</span>'
			                break;
			            case 'Cancelled':
			                status = '<span class="label label-grey arrowed-in-left arrowed-in">Cancelled</span>'
			                break;
			            case 'OnHold':
			                status = '<span class="label label-pink arrowed-right">OnHold</span>'
			                break;

			            default:
			                status = '<span class="label label-yellow  arrowed-right arrowed-in">' + value + '</span>';
			                break;
			        }
			        return "<div>" + status + "</div>";
			    }
			}
            ]

**Comment by olifolkerd** [comment]
I have just pushed a fix for this to the repo.

There was an issue in the row redraw function, where the index of columns was not being correctly set

Thanks for your patience.

Many Thanks.

**Comment by yaxino** [comment]
Great.
Is it possible for me to download the latest version of the code to solve this problem?

**Comment by olifolkerd** [comment]
yeah, just pull down the latest code from the repo, or if you are not using git click on the green "download" button towards the top right of the dashboard for this repo.

Cheers

Oli

**Comment by yaxino** [comment]
very well, 

let me try it .
